internal/bminventory/inventory.go

```diff
@@ -258,6 +258,55 @@ type bareMetalInventory struct {
 	secretValidator validations.PullSecretValidator
 }
 
+func (b *bareMetalInventory) UpdateClusterInstallProgress(ctx context.Context, params installer.UpdateClusterInstallProgressParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	var c common.Cluster
+
+	txSuccess := false
+
+	tx := b.db.Begin()
+	tx = transaction.AddForUpdateQueryOption(tx)
+
+	defer func() {
+		if !txSuccess {
+			log.Error("cluster installation progress update failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("cluster installation progress update failed")
+			tx.Rollback()
+		}
+	}()
+
+	if tx.Error != nil {
+		log.WithError(tx.Error).Errorf("failed to start db transaction")
+		return installer.NewUpdateClusterInstallProgressInternalServerError().WithPayload(
+			common.GenerateError(http.StatusInternalServerError, errors.New("DB error, failed to start transaction")))
+	}
+
+	if err := tx.Preload("Hosts").First(&c, "id = ?", params.ClusterID).Error; err != nil {
```

Comment:
`Preload("Hosts")` not really needed because you are not using the hosts.

Reply:
removed.